features/wallet/impl/src/main/java/com/tangem/feature/wallet/presentation/wallet/viewmodels/WalletNotificationsListFactory.kt

```diff
@@ -98,6 +111,33 @@ internal class WalletNotificationsListFactory(
         }
     }
 
+    private fun hasUnreachableNetworks(): Boolean {
+        val isUnreachableState = { item: WalletContentItemState.MultiCurrencyItem ->
+            (item as? WalletContentItemState.MultiCurrencyItem.Token)?.state is TokenItemState.Unreachable
+        }
+
+        return currentStateProvider().let { state ->
+            state is WalletStateHolder.MultiCurrencyContent && state.contentItems.any(isUnreachableState)
+        }
+    }
+
+    private fun TokenList.hasMissedDerivations(): Boolean {
+        val hasMissedDerivation: List<TokenStatus.Status>.() -> Boolean = { any { it is TokenStatus.MissedDerivation } }
+
+        val groupedListStatuses = (this as? TokenList.GroupedByNetwork)
+            ?.groups
+            ?.flatMap(NetworkGroup::tokens)
+            ?.map(TokenStatus::value)
+            .orEmpty()
+
+        val ungroupedListStatuses = (this as? TokenList.Ungrouped)
+            ?.tokens
+            ?.map(TokenStatus::value)
+            .orEmpty()
+
+        return groupedListStatuses.hasMissedDerivation() || ungroupedListStatuses.hasMissedDerivation()
```

Comment:
Кажется так попроще будет

```suggestion
        val tokens = when (this) { 
            // get tokens statuses from list
        }
        
        return tokens.any { it.value is TokenStatus.MissedDerivation }
```

Reply:
Крутое решение 👍

Reply:
Исправил